Bonobo.Git.Server/Git/GitLfs/ILfsDataStorageProvider.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.IO;
+
+namespace Bonobo.Git.Server.Git.GitLfs
+{
+    ///<summary>Represents a method of storage for LFS objects.</summary>
+    public interface ILfsDataStorageProvider
+    {
+        /// <summary>Produces the canonical URL for a file which LFS clients can use for subsequent LFS operations.</summary>
+        string GetFileUrl(string urlScheme, string urlAuthority, string requestApplicationPath, string operation, string repositoryName, string oid, long size);
```

Comment:
I do not believe that this method belongs with the storage provider. Rather than passing in all of this data so it can build you a path, why not let it pass back a relative path?

For that matter it seems like the physical path is what this provider would be dealing in. Later the LfsService would come along and translate it. The LfsService or Controller is in a much better position to formulate the URL without leaking that information into this class.

Reply:
Agreed.  I'm going to move it and change the way it works.

Reply:
This has been moved to the action provider.